lib/firebase_functions_interop.dart

```diff
@@ -378,3 +381,51 @@ class FirestoreEvent extends Event<DeltaDocumentSnapshot> {
           timestamp: timestamp,
         );
 }
+
+class PubsubFunctions {
+  const PubsubFunctions._();
+
+  TopicBuilder topic(String path) => new TopicBuilder._(_js.pubsub.topic(path));
+}
+
+class TopicBuilder {
+  @protected
+  final js.TopicBuilder nativeInstance;
+
+  TopicBuilder._(this.nativeInstance);
+
+  /// Event handler that fires every time an event is public in Pubsub.
+  js.CloudFunction onPublish(FutureOr<void> handler(Message message)) {
+    dynamic wrapper(js.Message jsMessage) => _handleEvent(jsMessage, handler);
+    return nativeInstance.onPublish(allowInterop(wrapper));
+  }
+
+  dynamic _handleEvent(
+      js.Message jsMessage, FutureOr<void> handler(Message message)) {
+    final Message message = new Message(jsMessage);
+    var result = handler(message);
+    if (result is Future) {
```

Comment:
I can't find anything in the docs about handlers being async, have you tried it?